resources/assets/js/annotations/annotatorContainer.vue

```diff
@@ -353,10 +353,10 @@ export default {
         handleDetachAnnotationLabel(annotation, annotationLabel) {
             if (this.isEditor) {
                 if (annotation.labels.length > 1) {
-                    AnnotationsStore.detachLabel(annotation, annotationLabel)
+                    return AnnotationsStore.detachLabel(annotation, annotationLabel)
                         .catch(handleErrorResponse);
```

Comment:
I think you only need this:

```suggestion
                    return AnnotationsStore.detachLabel(annotation, annotationLabel)
                        .then(() => this.refreshAnnotation(annotation))
                        .catch(handleErrorResponse);
```

This will refresh the annotation if the label is swapped *and* if a label is deleted (which currently still won't update the color). No need to return all the promises and update `handleSwapLabel()`.